chat-plugins/prowrestling.js

```diff
@@ -0,0 +1,209 @@
+/**
+ * Professional Wrestling Room Plugin
+ * Pokemon Showdown - http://pokemonshowdown.com/
+ *
+ * This handles submitting questions for the podcast that the room runs periodically,
+ * instead of having to use another source to ask questions.
+ *
+ * @license MIT license
+ */
+'use strict';
+
+const FS = require('../fs');
+
+const PW_DATA_PATH = 'config/chat-plugins/prowrestling-questions.json';
+const MAX_QUESTION_LENGTH = 500;
+
+class ProWrestling {
+	constructor() {
+		this.pwData = Object.create(null);
+		try {
+			this.pwData = require(`../${PW_DATA_PATH}`);
+		} catch (error) {
+			if (error.code !== 'MODULE_NOT_FOUND') throw error;
+		}
+
+		this.saveQuestions = (() => {
+			let writing = false;
+			let writePending = false;
+			return async () => {
+				if (writing) {
+					writePending = true;
+					return;
+				}
+				writing = true;
+
+				await FS(`${PW_DATA_PATH}.0`).write(JSON.stringify(this.pwData));
+				await FS(`${PW_DATA_PATH}.0`).rename(PW_DATA_PATH);
+
+				writing = false;
+				if (writePending) {
+					writePending = false;
+					setImmediate(() => this.saveQuestions());
+				}
+			};
+		})();
+	}
+	askQuestion(question, user) {
+		const num = Object.keys(this.pwData).length + 1;
+		this.pwData[num] = {
```

Comment:
Doing this on mobile but say there are 2 questions: 1 and 2.  if the first question is deleted i would assume that the capacity of this database would stay at 1 question since keys.length would always be 1

you would constantly be writing the n2 slot

Reply:
I'm going to update this later.

What I'm planning on doing is just ``userid#number`` - based on how many questions they've asked. (it'll store an array of userids and the number of questions they've asked long term)